src/base.jl

```diff
@@ -496,21 +496,57 @@ Initialise then begin a least squares fitting of the supplied data.
 - `singularity`: Presence of singularity in model
 - `indweight`: indices weighting, see `modelfit` for more discussion
 - `optmethod`: optimisation algorithm used by NLOpt. 
-"""
-function leastsquares_init(params_init::Vector{RheoFloat}, low_bounds::RheovecOrNone,
-                           hi_bounds::RheovecOrNone, modulus,
-                           time_series::Vector{RheoFloat}, dt::RheoFloat,
-                           prescribed_dot::Vector{RheoFloat}, measured::Vector{RheoFloat};
-                           insight::Bool = false, constant_sampling::Bool=true,
-                           singularity::Bool = false, _rel_tol = 1e-4, indweights=nothing,
-                           optmethod::Symbol = :LN_SBPLX)
+- `opttimeout`: allows user to set a wall clock timeout on optimisation 
+"""
+function leastsquares_init(params_init::Vector{RheoFloat},
+                            low_bounds::RheovecOrNone,
+                            hi_bounds::RheovecOrNone, 
+                            modulus,
+                            time_series::Vector{RheoFloat},
+                            dt::RheoFloat,
+                            prescribed_dot::Vector{RheoFloat},
+                            measured::Vector{RheoFloat};
+                            insight::Bool = false,
+                            constant_sampling::Bool=true,
+                            singularity::Bool = false,
+                            rel_tol::Union{Real,Nothing} = nothing,
+                            rel_tol_f::Union{Real,Nothing} = nothing,
+                            indweights = nothing,
+                            optmethod::Symbol = :LN_SBPLX,
+                            opttimeout::Union{Real,Nothing} = nothing,
+                            optmaxeval::Union{Integer,Nothing} = nothing)
                            
 
     # initialise NLOpt.Opt object with :LN_SBPLX Subplex algorithm
     opt = Opt(optmethod, length(params_init))
     # opt = Opt(:LN_BOBYQA, length(params_init))    # Passing tests
     # opt = Opt(:LN_COBYLA, length(params_init))    # Failing test - not precise enough?
 
+    # set optimiser stopping criteria
+
+    # wall clock timeout
+    if !isnothing(opttimeout)
+        opttimeout = convert(RheoFloat, opttimeout)
+        maxtime!(opt, opttimeout)
+    end
+
+    # evaluation cycle ceiling
+    if !isnothing(optmaxeval)
+        maxeval!(opt, optmaxeval)
+    end
+
+    # input parameter change tolerance
+    if !isnothing(rel_tol)
+        rel_tol = convert(RheoFloat, rel_tol)
+        xtol_rel!(opt, rel_tol)
+    end
+
+    # objective function change tolerance 
+    if !isnothing(rel_tol_f)
+        rel_tol = convert(RheoFloat, rel_tol_f)
```

Comment:
This doesn't look right. should this be `ftol_rel!` rather than `xtol_rel!` ?
There is the same code in the step version of the function.

Reply:
a howler from me there!